### PR TITLE
Add stackwalker

### PR DESCRIPTION
**Pull request by SirLynix**
From https://github.com/JochenKalmbach/StackWalker

### CODE REVIEW

File: packages/s/stackwalker/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    set_homepage("https://curl.haxx.se/")
+    set_description("The multiprotocol file transfer library.")
+
+    set_urls("https://github.com/JochenKalmbach/StackWalker/archive/$(version).zip",
+             "https://github.com/JochenKalmbach/StackWalker.git")
+    add_versions("1.20", "b139c83b7c4991930ebe48eae43b0feedca034e136f00be294f3641495b2c835")
+
+    add_deps("cmake")
+    add_syslinks("advapi32")
```
Comment (yamashi) [inline packages/s/stackwalker/xmake.lua:11]: Needs to be wrapper in windows platform check

File: packages/s/stackwalker/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    on_install("windows", function (package)
+        local configs = {"-DStackWalker_DISABLE_TESTS=ON"}
```
Comment (yamashi) [inline packages/s/stackwalker/xmake.lua:14]: add options for Release/Debug via CMAKE_BUILD_TYPE

**Review by yamashi — CHANGES_REQUESTED** [review]

File: packages/s/stackwalker/xmake.lua (hunk 1 above)
Comment (SirLynix) [reply packages/s/stackwalker/xmake.lua:11]: This is a Windows only library, is this really necessary?

File: packages/s/stackwalker/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package("stackwalker")
+
+    set_homepage("https://curl.haxx.se/")
```
Comment (yamashi) [inline packages/s/stackwalker/xmake.lua:3]: curl 👼

File: packages/s/stackwalker/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    set_description("The multiprotocol file transfer library.")
```
Comment (yamashi) [inline packages/s/stackwalker/xmake.lua:4]: curl description

File: packages/s/stackwalker/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    add_syslinks("advapi32")
+
+    on_install("windows", function (package)
+        local configs = {"-DStackWalker_DISABLE_TESTS=ON"}
+        import("package.tools.cmake").install(package, configs)
+   end)
+
+    on_test(function (package)
+        assert(package:check_cxxsnippets({test = [[
+            #include "StackWalker.h"
```
Comment (yamashi) [inline packages/s/stackwalker/xmake.lua:20]: includes should be in the config of this function

**Review by yamashi — CHANGES_REQUESTED** [review]

**Comment by waruqi** [comment]
Thanks. 👍

**Pull request merged**